Sources/PriorityQueueModule/Heap+Iterator.swift

```diff
@@ -0,0 +1,54 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Collections open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+extension Heap {
+  /// Adds sequence conformance to both the min and max view of the heap
+  public struct Iterator: Sequence, IteratorProtocol {
+
+    public enum IterationDirection {
+      case ascending
+      case descending
+    }
+
+    @usableFromInline
+    internal var _base: Heap
+
+    @usableFromInline
+    internal let _direction: IterationDirection
+
+    @inlinable
+    public init(_base: Heap, _direction:IterationDirection) {
```

Comment:
I don't _think_ there's any reason to make this initializer (or the `IterationDirection` enum) public. (It can be marked as `@inlinable` even if it's internal).